medusa/tv/series.py

```diff
@@ -1377,6 +1377,14 @@ def make_ep_from_file(self, filepath):
                     continue
 
             else:
+                # If the episodes status is set to archive, leave it alone.
```

Comment:
It's still a hypothesis, but I have a suspicion that is the reason that sometime the episode status switches from archived back to a downloaded status. And which will result it to be available for "upgrades".

I think it has something todo, with when something on the filesystem changes. Like maybe the file itself. And then it can be evaluated by the should_refresh method, which can also change it's status.

Shortcutting it, maybe not fix the the root cause. But I think it wil realy prevent this from happening again.

@fernandog @medariox can you think of situations where we don't want to shortcut this when the status is ARCHIVED?